net/Socket.cpp

```diff
@@ -1553,7 +1553,7 @@ bool StreamSocket::checkRemoval(std::chrono::steady_clock::time_point now)
                                                     << ", " << *this);
         if (_socketHandler)
         {
-            _socketHandler->onDisconnect();
```

Comment:
Is it possible that in the past we called `onDisconnect()` more than once?

It looks possible, given the code never checked (and we don't reset `_socketHandler` here). I hope we don't have side-effects to this that we now break in subtle ways when we call it exactly once.

Reply:
I think it is unlikely to be called >once - but sometimes is not called at all which is surprising - and I think this has bitten us lifecycle wise in the past in corner-cases. But yes - anything subtle I want to flush out and make explicit now in the 25.04 lifecycle - so lets get it in now & iterate  :-)